ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationPeriodValidationIT.scala

```diff
@@ -69,6 +115,116 @@ class CommandDeduplicationPeriodValidationIT extends LedgerTestSuite {
           )
         ),
       )
-    }
+      metadataLongestDuration = extractErrorInfoMetadataValue(failure, "longest_duration")
+      // request accepted and the metadata value is valid
+      _ <- ledger.submit(
+        request.update(
+          _.commands.deduplicationDuration := DurationConversion.toProto(
+            Duration.parse(metadataLongestDuration)
+          )
+        )
+      )
+    } yield {}
+  })
+
+  test(
+    "OffsetOutsideRange",
+    "Submission with deduplication periods represented by offsets which are outside the valid range are rejected",
+    allocate(SingleParty),
+    // Canton accepts the given offsets
+    enabled =
+      _.commandDeduplicationFeatures.getDeduplicationPeriodSupport.offsetSupport.isOffsetConvertToDuration,
+    disabledReason = "Only ledgers than convert offsets to durations fail",
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      end <- ledger.offsetBeyondLedgerEnd()
+      deduplicationPeriod = DeduplicationPeriod.DeduplicationOffset(
+        Ref.HexString.assertFromString(end.getAbsolute.toLowerCase)
+      )
+      _ <- assertFailedRequest(
+        ledger,
+        party,
+        deduplicationPeriod,
+        "Submitting a command with an invalid offset",
+        "The submitted command had an invalid deduplication period: Cannot convert deduplication offset to duration because there is no completion at given offset .*",
```

Comment:
```suggestion
        "Invalid deduplication period in the submitted command: Cannot convert deduplication offset to duration because there is no completion at given offset .*",
```

Reply:
Same point as previosly